src/utilities.cpp

```diff
@@ -457,4 +457,37 @@ size_t getVariableIndexInComponent(const ComponentPtr &component, const Variable
     return index;
 }
 
+bool isEntityChildOf(const EntityPtr &entity1, const EntityPtr &entity2)
+{
+    return entity1->parent() == entity2;
+}
+
+bool areEntitiesSiblings(const EntityPtr &entity1, const EntityPtr &entity2)
+{
+    auto entity1Parent = entity1->parent();
+    return entity1Parent != nullptr && entity1Parent == entity2->parent();
+}
+
+InterfaceTypePair determineInterfaceType(const VariablePtr &variable1, const VariablePtr &variable2)
+{
```

Comment:
should this function name just be interfaceType() to follow our other get/set format?

Reply:
It is not getting an interface type it is working out what the interface type for both variables should be.